phpBB/phpbb/textformatter/renderer.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace phpbb\textformatter;
+
+abstract class renderer
```

Comment:
Made an interface. I removed configure_smilies_path() and configure_user(), not sure whether that's should be a feature. At the moment it's just an implementation detail.